src/EditorFeatures/TestUtilities2/Intellisense/TestStateBase.vb

```diff
@@ -0,0 +1,304 @@
+                Assert.Equal(displayText, GetDisplayText(Me.CurrentSignatureHelpPresenterSession.SelectedItem, Me.CurrentSignatureHelpPresenterSession.SelectedParameter.Value))
+            End If
+
+            If documentation IsNot Nothing Then
+                Assert.Equal(documentation, Me.CurrentSignatureHelpPresenterSession.SelectedItem.DocumentationFactory(CancellationToken.None).GetFullText())
+            End If
+
+            If selectedParameter IsNot Nothing Then
+                Assert.Equal(selectedParameter, GetDisplayText(
+                    Me.CurrentSignatureHelpPresenterSession.SelectedItem.Parameters(
+                        Me.CurrentSignatureHelpPresenterSession.SelectedParameter.Value).DisplayParts))
+            End If
+        End Function
+#End Region
+
+#Region "Helpers"
+
+        Private Shared Function CombineExcludedTypes(excludedTypes As IList(Of Type), includeFormatCommandHandler As Boolean) As IList(Of Type)
+            Dim result = New List(Of Type) From {
+                GetType(IIntelliSensePresenter(Of ICompletionPresenterSession, ICompletionSession)),
+                GetType(IIntelliSensePresenter(Of ISignatureHelpPresenterSession, ISignatureHelpSession))
+            }
+
+            If Not includeFormatCommandHandler Then
+                result.Add(GetType(FormatCommandHandler))
+            End If
+
+            If excludedTypes IsNot Nothing Then
+                result.AddRange(excludedTypes)
+            End If
+
+            Return result
+        End Function
+
+        Private Shared Function CombineExtraTypes(extraExportedTypes As IList(Of Type)) As IList(Of Type)
+            Dim result = New List(Of Type) From {
+                GetType(TestCompletionPresenter),
+                GetType(TestSignatureHelpPresenter),
+                GetType(IntelliSenseTestState)
+            }
+
+            If extraExportedTypes IsNot Nothing Then
+                result.AddRange(extraExportedTypes)
+            End If
+
+            Return result
+        End Function
+
+        Private Shared Function GetDisplayText(item As SignatureHelpItem, selectedParameter As Integer) As String
+            Dim suffix = If(selectedParameter < item.Parameters.Count,
+                            GetDisplayText(item.Parameters(selectedParameter).SuffixDisplayParts),
+                            String.Empty)
+            Return String.Join(
+                String.Empty,
+                GetDisplayText(item.PrefixDisplayParts),
+                String.Join(
+                    GetDisplayText(item.SeparatorDisplayParts),
+                    item.Parameters.Select(Function(p) GetDisplayText(p.DisplayParts))),
+                GetDisplayText(item.SuffixDisplayParts),
+                suffix)
+        End Function
+
+        Private Shared Function GetDisplayText(parts As IEnumerable(Of TaggedText)) As String
+            Return String.Join(String.Empty, parts.Select(Function(p) p.ToString()))
+        End Function
```

Comment:
were these functions moved from somewhere?

Reply:
yes, the original class was TestState.vb. It is included in the CR as well. Here is the original location of those functions: [https://github.com/dotnet/roslyn/blob/ca0e683cb40bd126b40088f5b44ef68092797ab7/src/EditorFeatures/TestUtilities2/Intellisense/TestState.vb#L511](https://github.com/dotnet/roslyn/blob/ca0e683cb40bd126b40088f5b44ef68092797ab7/src/EditorFeatures/TestUtilities2/Intellisense/TestState.vb#L511)



---
In reply to: [232508236](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508236) [](ancestors = 232508236)